rpcs3/Emu/RSX/Overlays/overlay_fonts.cpp

```diff
@@ -294,128 +294,107 @@ namespace rsx
 				return;
 			}
 
-			usz i                = 0u;
-			bool skip_whitespace = false;
-
-			while (true)
+			// Render as many characters as possible as glyphs.
+			for (usz i = 0u, begin_of_word = 0u; i < char_limit; i++)
 			{
-				if (auto c = text[i++]; c && (i <= char_limit))
+				switch (const auto& c = text[i])
 				{
-					switch (c)
-					{
-					case '\n':
+				case '\0':
+				{
+					// We're done.
+					return;
+				}
+				case '\n':
+				{
+					// Reset x to 0 and increase y to advance to the new line.
+					x_advance = 0.f;
+					y_advance += size_px + 2.f;
+					begin_of_word = result.size();
+					continue;
+				}
+				case '\r':
+				{
+					// Reset x to 0.
+					x_advance = 0.f;
+					begin_of_word = result.size();
+					continue;
+				}
+				default:
+				{
+					const bool is_whitespace = c == ' ';
+					stbtt_aligned_quad quad{};
+
+					if (is_whitespace)
 					{
-						y_advance += size_px + 2.f;
-						x_advance = 0.f;
-						continue;
+						// Skip whitespace if we are at the start of a line.
+						if (x_advance > 0.f)
+						{
+							// Get the glyph size.
+							quad = get_char(c, x_advance, y_advance);
+
+							// Reset the result if the glyph would protrude out of the given space anyway.
+							if (x_advance > max_width)
+							{
+								quad = {};
+							}
+						}
 					}
-					case '\r':
+					else
 					{
-						x_advance = 0.f;
-						continue;
+						// No whitespace. Get the glyph size.
+						quad = get_char(c, x_advance, y_advance);
 					}
-					default:
+
+					// Add the glyph's vertices.
+					result.emplace_back(quad.x0, quad.y0, quad.s0, quad.t0);
+					result.emplace_back(quad.x1, quad.y0, quad.s1, quad.t0);
+					result.emplace_back(quad.x0, quad.y1, quad.s0, quad.t1);
+					result.emplace_back(quad.x1, quad.y1, quad.s1, quad.t1);
+
+					// The next word will begin after any whitespaces.
+					if (is_whitespace)
```

Comment:
You can have multiple whitespace sequence which is why there was a boolean flag tracking until the actual beginning is found.

Reply:
You mean multiple flags and a horrible reverse loop xD
All I know is this code works and the old one didn't.

Reply:
It does't if you have multiple spaces before a word.